src/mbgl/gl/context.cpp

```diff
@@ -124,10 +124,27 @@ void Context::initializeExtensions(const std::function<gl::ProcAddress(const cha
             return nullptr;
         };
 
-        debugging = std::make_unique<extension::Debugging>(fn);
-        if (!disableVAOExtension) {
-            vertexArray = std::make_unique<extension::VertexArray>(fn);
+        const std::string renderer = reinterpret_cast<const char*>(MBGL_CHECK_ERROR(glGetString(GL_RENDERER)));
+        Log::Info(Event::General, "GPU Identifier: %s", renderer.c_str());
+
+        // Block ANGLE on Direct3D since the debugging extension is causing crashes
+        if (renderer.find("ANGLE") == std::string::npos
+            || renderer.find("Direct3D") == std::string::npos) {
```

Comment:
While this may be the correct logic (disable Debugging for Angle on Direct3D), it's a bit confusing to read this code. I think

```cpp
!(renderer.find("ANGLE") != std::string::npos && renderer.find("Direct3D") != std::string::npos)
```

may be easier to read.

Reply:
:+1: I actually wrote that first and then started wondering if the double negation made it more confusing. It seems it doesn't. I'll change it.